api/v1alpha1/internalserviceexport_types.go

```diff
@@ -18,6 +19,19 @@ type InternalServiceExportSpec struct {
 	// The reference to the source Service.
 	// +kubebuilder:validation:Required
 	ServiceReference ExportedObjectReference `json:"serviceReference"`
+	// Type is the type of the Service in each cluster.
+	Type corev1.ServiceType `json:"type,omitempty"`
+	// IsDNSLabelConfigured determines if the Service has a DNS label configured.
+	// A valid "service.beta.kubernetes.io/azure-dns-label-name" should be configured when the public IP address of the
```

Comment:
There's chance that cx can provide a BYO public IP with dns label configured. In this case, they do not need to specify the annotation. I think we should use the configuration on Azure public IP resource as the ground truth.